File: uxmobile/src/main/java/sk/brecka/uxmobile/model/event/ViewEvent.java
package sk.brecka.uxmobile.model.event;

import org.json.JSONArray;
import org.json.JSONException;

import sk.brecka.uxmobile.model.ViewEnum;

/**
 * Created by matej on 4.10.2017.
 */

public abstract class ViewEvent extends TouchEvent {
    protected static final int INDEX_VIEW = 4;
    protected static final int INDEX_VIEW_TEXT = 5;
    protected static final int INDEX_VIEW_VALUE = 6;

    private final ViewEnum mViewEnum;
    private final String mViewText;
    private final String mViewValue;

    protected ViewEvent(long startTime, double x, double y, ViewEnum viewEnum, String viewText, String viewValue) {
        super(startTime, x, y);
        mViewEnum = viewEnum;
        mViewText = viewText;
        mViewValue = viewValue;
    }

    @Override
    public JSONArray toJson() throws JSONException {
        return super.toJson()
                .put(INDEX_VIEW, mViewEnum.getJsonValue())
                .put(INDEX_VIEW_TEXT, mViewText)
                .put(INDEX_VIEW_VALUE, mViewValue);
    }
}

File: uxmobile/src/main/java/sk/uxtweak/uxmobile/model/event/ViewEvent.kt
package sk.uxtweak.uxmobile.model.event

import org.json.JSONArray
import sk.uxtweak.uxmobile.model.ViewEnum

abstract class ViewEvent(
    startTime: Long,
    x: Float,
    y: Float,
    private val viewEnum: ViewEnum,
    private val viewText: String,
    private val viewValue: String
) : TouchEvent(startTime, x, y) {
    override fun toJson(): JSONArray = super.toJson()
        .put(INDEX_VIEW, viewEnum.jsonValue)
        .put(INDEX_VIEW_TEXT, viewText)
        .put(INDEX_VIEW_VALUE, viewValue)

    companion object {
        const val INDEX_VIEW = 4
        const val INDEX_VIEW_TEXT = 5
        const val INDEX_VIEW_VALUE = 6
    }
}
